docs/source/indexes/ivfflat.rst

```diff
@@ -0,0 +1,116 @@
+
+
+Configuration parameters
+------------------------
+
+Build parameters
+~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Name
+     - Default
+     - Description
+   * - n_lists
+     - sqrt(n)
+     - Number of coarse clusters used to partition the index. A good heuristic for this value is sqrt(n_vectors_in_index)
+   * - add_data_on_build
+     - True
+     - Should the training points be added to the index after the index is built?
+   * - kmeans_train_iters
+     - 20
+     - Max number of iterations for k-means training before convergence is assumed. Note that convergence could happen before this number of iterations.
+   * - kmeans_trainset_fraction
+     - 0.5
+     - Fraction of points that should be subsampled from the original dataset to train the k-means clusters. Default is 1/2 the training dataset. This can often be reduced for very large datasets to improve both cluster quality and the build time.
+   * - adaptive_centers
+     - false
+     - Should the existing trained centroids adapt to new points that are added to the index? This provides a trade-off between improving recall at the expense of having to compute new centroids for clusters when new points are added. When points are added in large batches, the performance cost may not be noticeable.
+   * - conservative_memory_allocation
+     - false
+     - To support dynamic indexes, where points are expected to be added later, the individual IVF lists can be imtentionally overallocated up front to reduce the amount and impact of increasing list sizes, which requires allocating more memory and copying the old list to the new, larger, list.
+
+
+Search parameters
+~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Name
+     - Default
+     - Description
+   * - n_probes
+     - 20
+     - Number of closest IVF lists to scan for each query point.
+
+Tuning Considerations
+---------------------
+
+Since IVF methods use clustering to establish spatial locality and partition data points into individual lists, there's an inherent
+assumption that the number of lists, and thus the max size of the data in the index is known up front. For some use-cases, this
+might not matter. For example, most vector databases build many smaller physical approximate nearest neighbors indexes, each from
+fixed-size or maximum-sized immutable segments and so the number of lists can be tuned based on the number of vectors in the indexes.
+
+Empirically, we've found :math:`\sqrt{n_index_vectors}` to be a good starting point for the :math:`n_lists` hyper-parameter. Remember, having more
+lists means less points to search within each list, but it could also mean more :math:`n_probes` are needed at search time to reach an acceptable
```

Comment:
```suggestion
lists means less points to search within each list, but it could also mean more :math:`n\_probes` are needed at search time to reach an acceptable
```